docs/src/Collage.doc.js

```diff
@@ -0,0 +1,270 @@
+// @flow
+import * as React from 'react';
+import stock1 from './images/stock1.jpg';
+import stock2 from './images/stock2.jpg';
+import stock3 from './images/stock3.jpg';
+import stock4 from './images/stock4.jpg';
+import stock5 from './images/stock5.jpg';
+import stock6 from './images/stock6.jpg';
+import PropTable from './components/PropTable.js';
+import Example from './components/Example.js';
+import PageHeader from './components/PageHeader.js';
+
+const cards = [];
+const card = c => cards.push(c);
+
+card(
+  <PageHeader
+    name="Collage"
+    description="
+Like Masonry, Collage creates a deterministic grid layout that can absolutely position and virtualize images.
+"
+  />
+);
+
+card(
+  <PropTable
+    props={[
+      {
+        name: 'height',
+        type: 'number',
+        description: 'Height of the collage',
+        required: true,
+      },
+      {
+        name: 'imageAlt',
+        type: 'string',
+        description: 'Accessibility label for every image',
+      },
+      {
+        name: 'images',
+        type:
+          'Array<{| color?: string, naturalHeight: number, naturalWidth: number, src: string |}>',
+        description: 'Images to show in the collage',
+        required: true,
+      },
+      {
+        name: 'imagesAreRounded',
+        type: 'boolean',
+        description: 'Use rounded corner for every image',
+      },
+      {
+        name: 'numCols',
+        type: 'number',
+        description: 'Number of columns',
+        required: true,
+      },
+      {
+        name: 'renderCoverImage',
+        type: '({ width: number, height: number }) => React.Node',
+        description: `Render prop for rendering the first image (a.k.a cover image).`,
+      },
+      {
+        name: 'space',
+        type: '{| type: "around" | "between", value: number |}',
```

Comment:
Seeing as both of these properties are required is there any value in having them be an object? If `value` changes we'd need to implement a custom `shouldComponentUpdate` to check the value of the `space` object. How about `gutterStyle: "around" | "between", gutterSize: number`?

Reply:
The reason I put it in an object is to make sure that either both `gutterSize` and `gutterStyle` are passed or none are passed. It doesn't make sense to require `gutterStyle` prop if `gutterSize` is 0 or undefined (which is the default).

Maybe we should make `gutterStyle` default to "between" and have both of them optional? Thoughts?

Also, this "space" value here is technically the padding of each image. The actual gutter size is 2 * image padding. That's why I didn't pick the name "gutter".

If we want to use "gutter" (which might be more intuitive), the parent could potentially pass a odd number and we might end up doing float point pixel values for the image padding, which I'm trying to avoid. Thoughts on this?

Reply:
On a second thought, padding gutter size makes more sense, especially now that we don't have the around case anymore. Will update to just accepting `gutter` prop